lib/engine/game/g_18_uruguay/nationalization.rb

```diff
@@ -23,60 +23,29 @@ def affected_shares(entity, corps)
           affected
         end
 
-        def find_president(holders, corps)
-          president_candidate = nil
-          candidate_sum = 0
-          holders.each do |holder|
-            entity_shares = affected_shares(holder, corps)
-            sum = entity_shares.sum(&:percent)
-            if sum > candidate_sum
-              president_candidate = holder
-              candidate_sum = sum
-            end
-          end
-          president_candidate
-        end
+        def acquire_shares
+          @merge_data[:holders].each do |holder|
+            entity_shares = affected_shares(holder, @merge_data[:corps])
 
-        def transfer_share(share, new_owner)
-          corp = share.corporation
-          corp.share_holders[share.owner] -= share.percent
-          corp.share_holders[new_owner] += share.percent
-          share.owner.shares_by_corporation[corp].delete(share)
-          new_owner.shares_by_corporation[corp] << share
-          share.owner = new_owner
-        end
-
-        def transfer_pres_share(corporation, owner)
-          pres_share = corporation.presidents_share
-          transfer_share(pres_share, owner)
-          corporation.owner = owner
-        end
-
-        def acquire_shares_in_fce(corp_fce, merge_data)
-          new_president = find_president(merge_data[:holders], merge_data[:corps])
-          transfer_pres_share(corp_fce, new_president)
-
-          merge_data[:holders].each do |holder|
-            aquired = 0
-            aquired = 20 if holder == new_president
-            entity_shares = affected_shares(holder, merge_data[:corps])
             total_percent = entity_shares.sum(&:percent)
-            aquire_percent = (total_percent / 20).to_i * 10
-            while aquired < aquire_percent
-              share = corp_fce.shares.first
-              aquired += share.percent
-              transfer_share(share, holder)
+            num_shares = (total_percent / 20).to_i
+            odd_share = num_shares * 20 != total_percent
+            from_secondary = @merge_data[:secondary_corps].reduce(0) do |sum, secondary_corps|
+              sum + (secondary_corps.president?(holder) ? 1 : 0)
             end
```

Comment:
You're trying to find how many of the secondary corporations the player is president of, right? This might be simpler:
```suggestion
            from_secondary = @merge_data[:secondary_corps].count { |corp| corp.president?(holder) }
```